apis/cpi/v1alpha1/cpiconfig_types.go

```diff
@@ -0,0 +1,205 @@
+
+	// Cluster CIDR
+	// +kubebuilder:validation:Optional
+	ClusterCidr string `json:"clusterCidr"`
+}
+
+type NsxT struct {
+	// A flag that enables pod routing
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	PodRoutingEnabled bool      `json:"podRoutingEnabled"`
+	Routes            NsxTRoute `json:"routes"`
+
+	// The username used to access NSX-T
+	// +kubebuilder:validation:Optional
+	Username string `json:"username"`
+
+	// The password used to access NSX-T
+	// +kubebuilder:validation:Optional
+	Password string `json:"password"`
+
+	// The NSX-T server
+	// +kubebuilder:validation:Optional
+	Host string `json:"host"`
+
+	// InsecureFlag is to be set to true if NSX-T uses self-signed cert
+	// +kubebuilder:validation:Optional
+	InsecureFlag bool `json:"insecureFlag"`
+
+	// RemoteAuth is to be set to true if NSX-T uses remote authentication (authentication done through the vIDM)
+	// +kubebuilder:validation:Optional
+	RemoteAuth bool `json:"remoteAuth"`
+
+	// VMCAccessToken is VMC access token for token based authentication
+	// +kubebuilder:validation:Optional
+	VmcAccessToken string `json:"vmcAccessToken"`
+
+	// VMCAuthHost is VMC verification host for token based authentication
+	// +kubebuilder:validation:Optional
+	VmcAuthHost string `json:"vmcAuthHost"`
+
+	// Client certificate key for NSX-T
+	// +kubebuilder:validation:Optional
+	ClientCertKeyData string `json:"clientCertKeyData"`
+
+	// Client certificate data for NSX-T
+	// +kubebuilder:validation:Optional
+	ClientCertData string `json:"clientCertData"`
+
+	// The certificate authority for the server certificate for locally signed certificates
+	// +kubebuilder:validation:Optional
+	RootCAData string `json:"rootCAData"`
+
+	// The name of secret that stores CPI configuration
+	// +kubebuilder:validation:Optional
+	SecretName string `json:"secretName"`
+
+	// The namespace of secret that stores CPI configuration
+	// +kubebuilder:validation:Optional
+	SecretNamespace string `json:"secretNamespace"`
+}
+
+type ExtraArgs struct {
+	// External arguments for cloud provider
+	// +kubebuilder:validation:Optional
+	TLSCipherSuites string `json:"tls-cipher-suites"`
+}
+
+type VSphereCPI struct {
+	// The vSphere mode. Either `vsphereCPI` or `vsphereParavirtualCPI`. Default value is `vsphereCPI`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=vsphereCPI
+	// +kubebuilder:validation:Enum=vsphereCPI;vsphereParavirtualCPI
+	Mode string `json:"mode"`
+
+	// The IP address or FQDN of the vSphere endpoint
+	// +kubebuilder:validation:Required
+	Server string `json:"server"`
+
+	// The datacenter in which VMs are created/located
+	// +kubebuilder:validation:Required
+	Datacenter string `json:"datacenter"`
+
+	// Username used to access a vSphere endpoint
+	// +kubebuilder:validation:Required
+	Username string `json:"username"`
+
+	// Password used to access a vSphere endpoint
+	// +kubebuilder:validation:Required
+	Password string `json:"password"`
+
```

Comment:
Looks like `region` and `zone` are missing

Reply:
Good catch. Adding it